music21/stream/base.py

```diff
@@ -607,7 +615,12 @@ def __getitem__(self,
 
             return t.cast(M21ObjType, searchElements[k])
 
-        elif isinstance(k, type) and issubclass(k, base.Music21Object):
+        elif all(
+            isinstance(maybeType, type) and issubclass(maybeType, base.Music21Object)
+            for maybeType in (
+                k if common.isIterable(k) else [k]  # type: ignore
+            )
+        ):
```

Comment:
not such a fan of this type of generator predicate.  Clearer to make two separate elif statements, one for the single class and second one for the isIterable case.  Students read this code and emulate it.